Structure_Engine/Query/Construction.cs

```diff
@@ -0,0 +1,111 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.ComponentModel;
+using BH.oM.Base.Attributes;
+using BH.oM.Physical.Constructions;
+using BH.oM.Structure.SurfaceProperties;
+using BH.oM.Quantities.Attributes;
+using BH.oM.Physical.Materials;
+
+namespace BH.Engine.Structure
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+        [Description("Creates a physical Construction from a structural ISurfaceProperty. Extracts the Structural MaterialFragment and creates a physical material with the same name.")]
+        [Input("surfaceProperty", "Structural surface property to convert.")]
+        [Output("construction", "The physical Construction to be used with ISurface such as Walls and Floors.")]
+        public static Construction IConstruction(this ISurfaceProperty surfaceProperty)
+        {
+            if (surfaceProperty.IsNull())
+                return null;
+            return Construction(surfaceProperty as dynamic);
+        }
+
+        //Write specific methods here if building a Construction from a SurfaceProperty via MaterialComposition is not valid
+        /***************************************************/
+        /**** Fallback Method                           ****/
+        /***************************************************/
+        [Description("Creates a physical Construction from a structural ISurfaceProperty. Extracts the Structural MaterialFragment and creates a physical material with the same name.")]
+        [Input("surfaceProperty", "Structural surface property to convert.")]
+        [Output("construction", "The physical Construction to be used with ISurface such as Walls and Floors.")]
+        public static Construction Construction(this ISurfaceProperty surfaceProperty)
+        {
+            if (surfaceProperty.IsNull())
+                return null;
+            MaterialComposition comp = surfaceProperty.IMaterialComposition();
+            double volume = surfaceProperty.IVolumePerArea();
+            double thickness = surfaceProperty.ITotalThickness();
+            List<oM.Physical.Constructions.Layer> layers = new List<oM.Physical.Constructions.Layer>();
+            for (int i = 0; i < comp.Materials.Count(); i++)
```

Comment:
Do you know whether the panel layers are ordered by any means? Can imagine insulation landing at the core of the wall 😃

Reply:
Actually now when I look at it, I would not involve `MaterialComposition` in this convert at all. Instead, I would rather work directly on `surfaceProperty.Layers` - it should be simpler plus voids could be maintained easily

Reply:
That's a good point- using MaterialComposition simplifies a lot for the other surface types, but I had left room for specific methods for types which needed it, and Layered would benefit from maintaining the order of void layers.

Reply:
Right, I see your point now. I have only worked with constant thickness and layered surfaces, so my perspective was too narrow. However, in this case I would consider dynamic casting with the following implementations:
- dedicated, simple method for constant thickness
- dedicated method for layered
- the method you created for everything else, with a warning saying that ribs etc. are being simplified and voids can be lost

Would that work?

Reply:
Yeah, I can add another for constant thickness. The warning comes up when volume < thickness, so I think there's nothing else needed there- I can re-word to specify that the void is a result of ribs/waffles/etc.